test/integration/suite.js

```diff
@@ -42,4 +42,11 @@ module.exports = function(knex) {
     });
   });
 
+  describe('knex.initialize', function() {
+    it('should allow initialize the pool with knex.initialize', function() {
+      knex.initialize();
```

Comment:
Isn't this missing pool destroy before trying to reinitialize? Also this is not actually waiting that initialization is done before testing that knex was initialized. 

Also check that before calling init and after calling destroy pool is actually destroyed.

Reply:
Hi, at that time of the test the pool is currently destroyed. I guess that you want the test to check first if the pool is initialized (maybe with !client.pool), and if not destroy it. The waiting until is destroyed is what i don't really know how to implement

Reply:
Looks like initialization is indeed synchronous opeartion, so that actually doesn't have to be waited. Yep, checking client.pool to not exist should be enough for testing that knex was not initialized at start of the test.